### PR TITLE
Updated member mrr_delta calculation to use stored mrr

### PR DESCRIPTION
**Pull request by SimonBackx**
refs https://github.com/TryGhost/Team/issues/1456

Simplifies the calculation of MRR deltas, which will make it easier to update MRR to include offers and cancellations in the future.

### CODE REVIEW

**Comment by SimonBackx** [comment]
https://github.com/TryGhost/Ghost/pull/14465 should be merged before this

**Comment by SimonBackx** [comment]
I'm going to include a solution for https://github.com/TryGhost/Team/issues/1451 in this PR as well, as this depends on these changes

**Comment by SimonBackx** [comment]
Tests in https://github.com/TryGhost/Ghost/pull/14470

File: packages/members-api/lib/repositories/member.js [hunk 1]
```diff
@@ -410,7 +423,8 @@ module.exports = class MemberRepository {
                     );
 
                     await this._StripeCustomerSubscription.upsert({
-                        status: updatedSubscription.status
+                        status: updatedSubscription.status,
+                        mrr: 0
```
Comment (allouis) [inline packages/members-api/lib/repositories/member.js:427]: We can delete all of this at the end of the cycle FYI. When a member is deleted so are all of the related subscriptions/events/relations

File: packages/members-api/lib/repositories/member.js [hunk 2]
```diff
@@ -712,10 +724,9 @@ module.exports = class MemberRepository {
                 id: model.id
             });
 
-            if (model.get('plan_id') !== updated.get('plan_id') || model.get('status') !== updated.get('status') || model.get('cancel_at_period_end') !== updated.get('cancel_at_period_end')) {
-                // TODO: we should use the previously saved MRR in the future as soon as the MRR column is 'stable' and migrated correctly
-                const originalMrrDelta = this.getMRR({interval: model.get('plan_interval'), amount: model.get('plan_amount'), status: model.get('status'), canceled: model.get('cancel_at_period_end')});
-                const updatedMrrDelta = this.getMRR({interval: updated.get('plan_interval'), amount: updated.get('plan_amount'), status: updated.get('status'), canceled: updated.get('cancel_at_period_end')});
+            if (model.get('mrr') !== updated.get('mrr') || model.get('plan_id') !== updated.get('plan_id') || model.get('status') !== updated.get('status') || model.get('cancel_at_period_end') !== updated.get('cancel_at_period_end')) {
```
Comment (allouis) [inline packages/members-api/lib/repositories/member.js:727]: It's worth considering simplifying this conditional to check just mrr and status, but it's not worth doing it now!

File: packages/members-api/lib/repositories/member.js (hunk 1 above)
Comment (SimonBackx) [reply packages/members-api/lib/repositories/member.js:427]: Is this already the case of will this change in 5.0?

File: packages/members-api/lib/repositories/member.js (hunk 2 above)
Comment (SimonBackx) [reply packages/members-api/lib/repositories/member.js:727]: Yes, you are right! I don't mind to do it now 😊

Comment (SimonBackx) [reply packages/members-api/lib/repositories/member.js:727]: We still need to include `cancel_at_period_end` until the dashboardv5 flag is dropped

**Comment by SimonBackx** [comment]
@allouis Could you do a full review? This should be ready to merge (after/with https://github.com/TryGhost/Ghost/pull/14465 and together with https://github.com/TryGhost/Ghost/pull/14470 after the version bump)

File: packages/members-api/lib/repositories/member.js (hunk 1 above)
Comment (allouis) [reply packages/members-api/lib/repositories/member.js:427]: It's already the case, but let's leave it for cleanup IMO!

**Review by allouis — CHANGES_REQUESTED** [review]
This looks good but I think we should revert the change to the conditional, I only commented as something to think about in future.

There are cases where we wouldn't create events, and we'd need to check with product whether or not we need/want an event for that.

One example is if you don't give a discount with yearly pricing, e.g. 10/month & 120/year - in this case upgrading/downgrading would not produce an event, and I'm not sure if that's desirable.

Other than that this looks 👍

**Review by allouis — APPROVED** [review]

**Pull request merged**